modules/doc/content/application_development/profiling.md

```diff
@@ -1,6 +1,177 @@
 # Profiling MOOSE code
 
-## MacOS
+For the deepest investigation into performance characteristics, we recommend
+using Google's [gperftools](https://github.com/gperftools/gperftools) package.
+This package is included as part of the environment package from the MOOSE
+installation process.  Although it works best on Linux platforms, it also
+works reasonably well on Mac OS.  Instruments also works well for profiling
+applications on Mac systems.
+
+## Google Performance Tools (Linux, Mac)
+
+Moose has support for profiling with
+[gperftools](https://github.com/gperftools/gperftools) built-in.  To use it,
+you must compile MOOSE with profiling support enabled.  If compiling with the
+latest MOOSE environment package, your applications will already be compiled
+with profiling support automatically. To manually control profiling support
+you set the GPERF_DIR environment variable to the location of a gperftools
+install (i.e. $GPERF_DIR/lib/libprofiler.so should exist).  It is also
+recommended you compile MOOSE in `oprof` mode to get complete/accurate
+profiling results.  Then you compile MOOSE like normal - it should look
+something like this:
+
+```
+export GPERF_DIR=/opt/moose/gperftools-2.7   # this should be unnecessary with the latest MOOSE package.
+export METHOD=oprof
+cd [your-moose-app-repository]
+make
+```
+
+This will compile your application with gperftools profiling support enabled.
+You are welcome to set GPERF_DIR to the location of your own gperftools
+installation as well.
+
+### Profiling
+
+To profile your application, you will need to run a simulation of suitable
+duration - generally at least a few seconds long - while setting the
+CPUPROFILE or MOOSE_PARALLEL_PROFILE environment variables (each for serial or
```

Comment:
This seems overly complicated to me. I would suggest that we drop mentioning CPUPROFILE in the docs at all. That way users don't have to think about serial versus parallel variables that aren't related in any logical way. Let's just use MOOSE_PARALLEL_PROFILE everywhere. I realize that the word PARALLEL is in there. We could update MOOSE and change this variable to MOOSE_PROFILE_FILE_BASE or something. Our users are very familiar with "file_base" at this point.

Reply:
I updated it to MOOSE_PROFILE_BASE because I couldn't bring myself to do ...PROFILE_FILE...